src/ripple/app/misc/tests/AmendmentTable.test.cpp

```diff
@@ -0,0 +1,391 @@
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+
+#include <ripple/app/misc/AmendmentTable.h>
+#include <ripple/core/ConfigSections.h>
+#include <beast/unit_test/suite.h>
+
+namespace ripple
+{
+class AmendmentTable_test final : public beast::unit_test::suite
+{
+public:
+    using StringPairVec = std::vector<std::pair<std::string, std::string>>;
+
+private:
+    enum class TablePopulationAlgo
+    {
+        addInitial,
+        addKnown
+    };
+
+    // 204/256 about 80%
+    static int const majorityFraction{204};
+
+    static void populateTable (AmendmentTable& table,
+                               std::vector<std::string> const& configLines)
+    {
+        Section section (SECTION_AMENDMENTS);
+        section.append (configLines);
+        table.addInitial (section);
+    }
+
+    static std::vector<AmendmentName> getAmendmentNames (
+        StringPairVec const& amendmentPairs)
+    {
+        std::vector<AmendmentName> amendmentNames;
+        amendmentNames.reserve (amendmentPairs.size ());
+        for (auto const& i : amendmentPairs)
+        {
+            amendmentNames.emplace_back (i.first, i.second);
+        }
+        return amendmentNames;
+    }
+
+    std::vector<AmendmentName> populateTable (
+        AmendmentTable& table,
+        StringPairVec const& amendmentPairs,
+        TablePopulationAlgo populationAlgo = TablePopulationAlgo::addKnown)
+    {
+        std::vector<AmendmentName> const amendmentNames (
+            getAmendmentNames (amendmentPairs));
+        switch (populationAlgo)
+        {
+            case TablePopulationAlgo::addKnown:
+                for (auto const& i : amendmentNames)
+                {
+                    table.addKnown (i);
+                }
+                break;
+            case TablePopulationAlgo::addInitial:
+            {
+                std::vector<std::string> configLines;
+                configLines.reserve (amendmentPairs.size ());
+                for (auto const& i : amendmentPairs)
+                {
+                    configLines.emplace_back (i.first + " " + i.second);
+                }
+                populateTable (table, configLines);
+            }
+            break;
+            default:
+                fail ("Error in test case logic");
+        }
+
+        return amendmentNames;
+    }
+
+    static std::unique_ptr<AmendmentTable> makeTable ()
+    {
+        beast::Journal journal;
+        return make_AmendmentTable (
+            weeks (2),
+            majorityFraction,
+            journal,
```

Comment:
If we got rid of the final bool parameter - as in your template design - that makes sense. Since I'm keeping the final parameter (for now) we don't gain anything by making journal a default parameter.